src/operator/controllers/external_traffic/endpoints_reconciler.go

```diff
@@ -130,7 +132,8 @@ func (r *EndpointsReconciler) InitIngressReferencedServicesIndex(mgr ctrl.Manage
 		v1alpha2.IngressServiceNamesIndexField,
 		func(object client.Object) []string {
 			ingress := object.(*v1.Ingress)
-			return serviceNamesFromIngress(ingress).List()
+			services := serviceNamesFromIngress(ingress)
+			return sets.List(services)
```

Comment:
On Go 1.19 the sets.Strings package is deprecated, this change upgrades to the new generic Set.